lib/server/server-middleware.js

```diff
@@ -98,6 +99,25 @@ const configureMiddleware = (options = {}) => {
     middleware.push(options.postStaticRoutes())
   }
 
+  const maintenanceUrl = '/restricted/maintenance'
+  if (MAINTENANCE_MODE) {
+    middleware.push((req, res, next) => {
+      if (req.originalUrl === maintenanceUrl) {
+        next()
+      } else {
+        res.redirect(maintenanceUrl)
+      }
+    })
+  } else {
+    middleware.push((req, res, next) => {
+      if (req.originalUrl === maintenanceUrl) {

```

Comment:
do we need to allow for css and json endpoints?

Reply:
It is done by inserting middleware of the static routes before the maintenance middleware here: https://github.com/ministryofjustice/fb-runner-node/blob/b199e03ec2ad87f932b5417f6662632e3e610040/lib/server/server-middleware.js#L84